### PR TITLE
🤽 `worker-sys` Make all methods `catch`

### PR DESCRIPTION
**Pull request by Jasper-Bekkers**
This is quite a big PR trying to address https://github.com/cloudflare/workers-rs/issues/545 that should serve as a starting point/discussion point wrt to using `catch` project wide in all `method`, `constructor` and `static_method_of` calls.

This is a blanket replacement and fixup of all calls into `worker-sys` without changing the higher level API. This means that we introduce a few `.unwrap` calls and that I'll piggyback on methods that already were returning a `Result` so we can `?` unwrap all the errors.

Going forward, if we choose to go down this route, I'd suggest enforcing this as a project-wide rule.

### CODE REVIEW

**Comment by Jasper-Bekkers** [comment]
> My only concern are the unwraps. I suppose this is still strictly better than before because the panic will contain more useful error messages, but I do wonder if we should bite the bullet and make the relevant methods all return Result.

Yeah my thought initially was that I don't want to change the public facing API, I wanted to leave that up to the project maintainers. I think the unwraps are fine in the places where they are since they're mostly in getters and like you say they'll contain relevant information rather then swallowing the error messages.

**Review by kflansburg — APPROVED** [review]

**Pull request merged**
